hail/src/main/scala/is/hail/backend/service/ServiceBackend.scala

```diff
@@ -186,21 +191,15 @@ class ServiceBackend(
 
     log.info(s"parallelizeAndComputeWithIndex: $token: running job")
 
-    val batchId = batchClient.create(
-      JObject(
-        "billing_project" -> JString(backendContext.billingProject),
-        "n_jobs" -> JInt(n),
-        "token" -> JString(token),
-        "attributes" -> JObject("name" -> JString(name + "_" + batchCount))),
-      jobs)
-
-    val batch = batchClient.waitForBatch(batchId)
-    batchCount += 1
+    totalNumWorkerJobs += n
+    val updateId = batchClient.update(batchId, token, jobs)
+    // The driver is in this batch too
+    val batch = batchClient.waitForBatch(batchId, totalNumWorkerJobs)
+    updateCount += 1
     implicit val formats: Formats = DefaultFormats
-    val batchID = (batch \ "id").extract[Int]
     val batchState = (batch \ "state").extract[String]
-    if (batchState != "success") {
-      throw new HailBatchFailure(s"$batchID")
+    if (batchState == "failed") {
```

Comment:
Shouldn't we handle `"cancelled"` as well? Let's also keep the batch id in the message.

Reply:
If the batch is `cancelled`, the job that's running this code would get cancelled as well, in which case it's not guaranteed that we'd even reach that point in the code. The python client controlling this QoB should be the one to pick up on the cancellation. AFAIK, the only states that the Query Driver can reliably observe here are failed and running.